### PR TITLE
[DS-2566] Add explore and view for history sidebar

### PR DESCRIPTION
**Pull request by m-d-bowerman**
Adds a view and an explore based on that view that includes fields related to history and bookmark sidebar interactions.

Checklist for reviewer:

When adding a new derived dataset:
- [ ] Ensure that the data is not available already (fully or partially) and recommend extending an existing dataset in favor of creating new ones. Data may be available in [bigquery-etl repository](https://github.com/mozilla/bigquery-etl), [looker-hub](https://github.com/mozilla/looker-hub) or in [looker-spoke-default](https://github.com/mozilla/looker-spoke-default/tree/e1315853507fc1ac6e78d252d53dc8df5f5f322b).
- [ ] Avoid merging a PR that includes the logic of a [core metric](https://docs.telemetry.mozilla.org/metrics/index.html) or complex business logic. The recommendation is to implement core business logic in bigquery-etl. E.g. The [type of search](https://github.com/mozilla/bigquery-etl/blob/a3e59f90326816a2ecaaa3e9d5b57fe9552f7d70/sql/moz-fx-data-shared-prod/search_derived/mobile_search_clients_daily_v1/query.sql#L781) or the [calculation of DAU or visited URIs](https://github.com/mozilla/bigquery-etl/blob/9bca48821a8a0d40b1700cc14ecd8068d132ed06/sql/moz-fx-data-shared-prod/telemetry_derived/firefox_desktop_exact_mau28_by_dimensions_v1/query.sql).
- [ ] Avoid merging code in Looker Explores/Views that implement analysis with multiple lines of code or that will be likely replicated in the future. Instead, aim for extending an existing dataset to include the required logic, and use [Looker aggregates](https://cloud.google.com/looker/docs/aggregate_awareness) to facilitate the analysis.
- [ ] Avoid merging a PR with logic that requires validation and health checks. It is recommended to implement it in bigquery-etl for full test coverage and failure alerts.

### CODE REVIEW

File: firefox_desktop/views/history_sidebar_clients_daily.view.lkml [hunk 1]
```diff
@@ -0,0 +1,86 @@
+view: history_sidebar_clients_daily {
+
+  sql_table_name: `moz-fx-data-shared-prod`.telemetry_derived.clients_daily_joined_v1 ;;
```
Comment (irrationalagent) [inline firefox_desktop/views/history_sidebar_clients_daily.view.lkml:3]: is there a reason you are referencing the _v1 version of this table? we usually use the views where possible, e.g. `mozdata.telemetry.clients_daily`

Also, we have in the past preferred the extension of things that are already in looker-hub, [and it looks like there is already a view for clients_daily_joined here](https://github.com/mozilla/looker-hub/blob/main/firefox_desktop/views/clients_daily_joined_table.view.lkml).

[Here is an example of how that kind of extension can work](https://github.com/mozilla/looker-spoke-default/blob/main/firefox_desktop/views/clients_daily_search.view.lkml#L1-L4). 

[Here are docs on extending views](https://cloud.google.com/looker/docs/reference/param-view-extends)

I could also be missing something here - is there is a reason why you need to have it the way you do?

**Review by irrationalagent — CHANGES_REQUESTED** [review]
This seems like it could work in theory, but I'd like to understand the reasoning for not extending the existing views as I mention in the other comment. 

Also, did you try this out on a branch in looker development mode to make sure that it works to  build the types of charts you need? I notice you don't have any client count measures here, so you are just looking for e.g. "total number of sidebar opens for the whole population"? rather than per-client numbers? Or maybe you are planning to join it to another explore to do that.

File: firefox_desktop/views/history_sidebar_clients_daily.view.lkml [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+  dimension: client_id {
+    type: string
+    sql: ${TABLE}.client_id ;;
+    hidden: yes
+  }
+
+  dimension: submission_date {
```
Comment (irrationalagent) [inline firefox_desktop/views/history_sidebar_clients_daily.view.lkml:11]: you can optionally extend this field to allow for timeframes other than dates, if you want.

[here is an example](https://github.com/mozilla/looker-hub/blob/main/firefox_desktop/views/clients_daily_joined_table.view.lkml#L1968-L1981)f

File: firefox_desktop/views/history_sidebar_clients_daily.view.lkml (hunk 1 above)
Comment (m-d-bowerman) [reply firefox_desktop/views/history_sidebar_clients_daily.view.lkml:3]: I didn't want all of the fields brought in from the `clients_daily_joined_table` view, which is why I was referencing the SQL table. I've changed it to extend the view, though -- having those fields in the future is probably worthwhile anyways.

File: firefox_desktop/views/history_sidebar_clients_daily.view.lkml (hunk 2 above)
Comment (m-d-bowerman) [reply firefox_desktop/views/history_sidebar_clients_daily.view.lkml:11]: Thanks! Added

File: firefox_desktop/views/history_sidebar_clients_daily.view.lkml (hunk 1 above)
Comment (irrationalagent) [reply firefox_desktop/views/history_sidebar_clients_daily.view.lkml]: OK I see - there are a lot of fields in the source table, and it seems like you don't need too many of them. 

[You can always hide the ones you don't want](https://community.looker.com/lookml-5/hiding-fields-from-an-extended-view-2755?postid=7310#post7310) - all you have to do is set the `hidden` field to `yes` for the fields that you don't want to include from the source view. 

Alternatively, if you feel that makes the lookml too bloated (many lines just to hide each field; I don't think there's another way) then I would be OK with you just referencing the table directly (though the view, not the materialized version like you originally had).

I'll leave it up to you.

**Review by irrationalagent — APPROVED** [review]

**Pull request merged**